state-synchronizer/src/logging.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{chunk_request::GetChunkRequest, chunk_response::GetChunkResponse};
+use anyhow::Error;
+use libra_config::config::PeerNetworkId;
+use libra_logger::Schema;
+use libra_types::{
+    contract_event::ContractEvent, ledger_info::LedgerInfoWithSignatures, waypoint::Waypoint,
+};
+use serde::Serialize;
+#[derive(Clone, Schema)]
+pub struct LogSchema<'a> {
+    name: LogEntry,
+    event: Option<LogEvent>,
+    #[schema(debug)]
+    error: Option<&'a Error>,
+    #[schema(display)]
+    peer: Option<&'a PeerNetworkId>,
+    is_upstream_peer: Option<bool>,
+    #[schema(display)]
+    chunk_request: Option<GetChunkRequest>,
+    version: Option<u64>,
+    #[schema(display)]
+    chunk_response: Option<GetChunkResponse>,
+    #[schema(display)]
+    waypoint: Option<Waypoint>,
+    subscription_name: Option<String>,
+    count: Option<usize>,
+    reconfig_events: Option<Vec<ContractEvent>>,
+    local_li_version: Option<u64>,
+    local_synced_version: Option<u64>,
+    local_epoch: Option<u64>,
+    #[schema(display)]
+    ledger_info: Option<LedgerInfoWithSignatures>,
+    old_epoch: Option<u64>,
+    new_epoch: Option<u64>,
+    target_version: Option<u64>,
+}
+
+impl<'a> LogSchema<'a> {
+    pub fn new(name: LogEntry) -> Self {
+        Self {
+            name,
+            event: None,
+            peer: None,
+            is_upstream_peer: None,
+            error: None,
+            chunk_request: None,
+            chunk_response: None,
+            version: None,
+            waypoint: None,
+            subscription_name: None,
+            reconfig_events: None,
+            count: None,
+            local_li_version: None,
+            local_synced_version: None,
+            local_epoch: None,
+            ledger_info: None,
+            old_epoch: None,
+            new_epoch: None,
+            target_version: None,
+        }
+    }
+
+    pub fn event_log(name: LogEntry, event: LogEvent) -> Self {
+        Self {
+            name,
+            event: Some(event),
+            peer: None,
+            is_upstream_peer: None,
+            error: None,
+            chunk_request: None,
+            chunk_response: None,
+            version: None,
+            waypoint: None,
+            subscription_name: None,
+            reconfig_events: None,
+            count: None,
+            local_li_version: None,
+            local_synced_version: None,
+            local_epoch: None,
+            ledger_info: None,
+            new_epoch: None,
+            old_epoch: None,
+            target_version: None,
+        }
+    }
```

Comment:
recommend you have a new_event or something that takes in name: LogEntry and event: Option<LogEvent> and both of these call into that... this avoid having 2x20 lines of code and two places that need to be updated if you add another field.